packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -601,54 +461,43 @@ const DataGrid = injectIntl((props) => {
-        onFocus={onFocus}
-        onBlur={onBlur}
-        onMouseDown={onMouseDown}
-        tabIndex={0}
-        {...(activeIndex != null && { onMouseUp, onMouseMove, onMouseLeave: onMouseUp })}
-      >
-        <ColumnContext.Provider
-          value={columnContextValue}
-        >
-          <ColumnHeader
-            columns={dataGridColumns}
-            headerHeight={columnHeaderHeight}
-            tableHeight={tableHeight}
-            activeColumnIndex={(gridHasFocus && focusedRow === 0) ? focusedCol : undefined}
-            isActiveColumnResizing={focusedRow === 0 && checkResizable}
-            columnResizeIncrement={columnResizeIncrement}
-            onColumnSelect={handleColumnSelect}
-            onResizeMouseDown={onResizeMouseDown}
-            onResizeHandleChange={onResizeHandleChange}
-          />
-          <tbody>
-            {rows.map((row, index) => (
-              <Row
-                rowIndex={index + 1}
-                key={row.id}
-                height={rowHeight}
-                id={row.id}
-                isSelected={row.isSelected}
-                cells={row.cells}
-                ariaLabel={row.ariaLabel}
-                hasRowSelection={hasSelectableRows}
-                displayedColumns={displayedColumns}
-                rowHeaderIndex={rowHeaderIndex}
-                onCellSelect={handleCellSelection}
-              />
-            ))}
-          </tbody>
-        </ColumnContext.Provider>
-      </table>
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+      tabIndex={0}
+      ref={gridContainerRef}
+      onKeyDown={handleKeyDown}
+      onMouseDown={onMouseDown}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      id={id}
+      className={cx('data-grid-container')}
+    >
+      <GridContext.Provider value={{ role: GridConstants.GRID, setCellAriaLiveMessage, setColumnHeaderAriaLiveMessage }}>
+        <Table
+          id={`${id}-table`}
+          rows={dataGridRows}
+          ariaLabelledBy={ariaLabelledBy}
+          ariaLabel={ariaLabel}
+          activeColumnIndex={(gridHasFocus && focusedRow === 0) ? focusedCol : undefined}
+          isActiveColumnResizing={focusedRow === 0 && checkResizable}
+          columnResizeIncrement={columnResizeIncrement}
+          pinnedColumns={pinnedColumns}
+          overflowColumns={overflowColumns}
+          defaultColumnWidth={defaultColumnWidth}
+          columnHeaderHeight={columnHeaderHeight}
+          rowHeight={rowHeight}
+          rowHeaderIndex={rowHeaderIndex}
+          onColumnResize={onColumnResize}
+          onColumnSelect={handleColumnSelect}
+          onCellSelect={handleCellSelection}
+          onRowSelectionHeaderSelect={handleRowSelectionHeaderSelect}
+          hasSelectableRows={hasSelectableRows}
+          isStriped
+          gridRef={grid}
+          gridContainerRef={tableContainerRef}
```

Comment:
You should probably just use the React Context Provider value if using this approach.  Then you don't require the props to be passed.